optimistix/_solver/quasi_newton.py

```diff
@@ -115,12 +78,615 @@ def __call__(
         The updated Hessian (or Hessian-inverse) approximation.
         """
 
+    @abc.abstractmethod
+    def init(self, y: Y, f: Scalar, grad: Y) -> tuple[_Hessian, HessianUpdateState]:
+        """
+        Initialize the Hessian structure and Hessian update state.
+
+        Set up a template structure of the Hessian to be used (with dummy values), as
+        well as the state of the update method, which stores a history of gradients for
+        limited-memory Hessian approximations.
+        """
+        pass
+
+
+class _LBFGSInverseHessianUpdateState(eqx.Module, Generic[Y]):
+    """
+    State variables for Algorithm 4.1 of:
+
+        Byrd, R. H., Nocedal, J., & Schnabel, R. B. (1994).
+        "Representations of quasi-Newton matrices and their use in limited memory
+        methods." *Mathematical Programming*, 63(1), 129–156.
+
+    **Arguments**:
+        - `index_start`: Index of the most recent update in the circular buffer.
```

Comment:
Nit, markdown needs a newline between these two lines to format bulleted lists correctly. (+the list should be dedented).

Not super important as IIUC this particular docstring isn't being pulled in and rendered in our docs.